openassessment/data.py

```diff
@@ -524,7 +527,20 @@ def collect_ora2_responses(cls, course_id, desired_statuses=None):
         else:
             statuses = AssessmentWorkflow().STATUS_VALUES
 
-        items = AssessmentWorkflow.objects.filter(course_id=course_id, status__in=statuses).values('item_id', 'status')
+        items_qs = AssessmentWorkflow.objects.filter(course_id=course_id, status__in=statuses)
+        if selected_cohort:
+            anonymous_user_ids = (CohortMembership.objects.
+                  filter(course_id=CourseKey.from_string(course_id), course_user_group__id=selected_cohort).
+                  values('user__anonymoususerid__anonymous_user_id')
+            )
+            relevant_submissions_uuid = (Submission.objects.
+                filter(student_item__student_id__in=anonymous_user_ids).
+                values('uuid')
```

Comment:
We can use `.values_list('uuid', flat=True)`. In this case `relevant_submissions_uuid_list` on L540 looks redundant.